src/public-assets/js/dropzone-image-upload.js

```diff
@@ -0,0 +1,91 @@
+var DropzoneImageUpload = (function ($, Notification, dropzoneOptions) {
+    var params = {
+        dropzoneTarget: $('#dropzone_media'),
+        urlInput: $('#' + dropzoneOptions.urlInputId),
+        removeImageLink: $('#remove_image_link'),
+        uploadInfoHolder: $('#dropzone_upload_info'),
+        events: ['click']
+    };
+
+    Dropzone.autoDiscover = false;
+
+    var
+        mediaUrl = params.urlInput,
+        removeEditMediaLink = params.removeImageLink,
+        removeEditMediaLinkParent = removeEditMediaLink.parent(),
+        dropzoneTarget = params.dropzoneTarget,
+        dropzoneMessageElement = dropzoneTarget.find('.dz-message');
+
+    if (mediaUrl.val()) {
+        removeEditMediaLinkParent.removeClass('hide');
+        removeEditMediaLink.on(params.events.click, function (event) {
+            event.preventDefault();
+            mediaUrl.val('');
+            activate();
+        });
+    }
+    else {
+        activate();
+    }
+
+    function activate() {
+        removeEditMediaLinkParent.addClass('hide');
+        dropzoneTarget.removeClass('hide');
+
+        var options = $.extend({
+            thumbnailWidth: null,
+            thumbnailHeight: null,
+            init: function () {
+                this.on('addedfile', function (file) {
```

Comment:
Hey @yemexx1 I think we can extract these functions as properties of this object so they can be reused if the init method is overwritten